controllers/front/validateBnpl.php

```diff
@@ -130,4 +145,49 @@ protected function initDisplayController()
     {
         return (_PS_VERSION_ < '1.5') ? new BWDisplay() : new FrontController();
     }
+
+    public function displayAjaxCreateOrder()
+    {
+        $return = [
+            'success' => false,
+            'idOrder' => null
+        ];
+        $response = $this->client->execute(new OrdersCreateRequest(new OrderBuilder($this->context)));
+
+        if ($response->getCode() < 300 && $response->getCode() > 199) {
+            $return['success'] = true;
+
+            if ($response instanceof HttpJsonResponse) {
+                $order = $response->toArray();
+
+                if (false === empty($order['id'])) {
+                    $return['idOrder'] = $order['id'];
+                }
+            }
+        }
+
+        die(json_encode($return));
+    }
+
+    public function displayAjaxCaptureOrder()
+    {
+        $order = json_decode(Tools::getValue('order'), true);
+        $return = [
+            'success' => false,
+        ];
+
+        if (false === empty($order['orderID'])) {
```

Comment:
Could you please add a validator on $order['orderID'] before using it ?
string [ 1 .. 36 ] characters https://developer.paypal.com/docs/api/orders/v2/#orders_get

Reply:
Validation added, the string must be validate HTML and less than 36 caracters.